### PR TITLE
DEPR #16747: deprecate pd.TimeGrouper 

### LINKED ISSUE #16747: DEPR: pd.TimeGrouper
#### Problem description
The `TimeGrouper` docstrings are missing the `key` and `level` parameters. `TimeGrouper` is also absent from the API reference page in the docs. I think it would also be nice to have a section in the docs on how it differs than `resample`.

### PR DESCRIPTION
**Pull request by stevenvandenberghe**
- [x] closes #16747
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master -u -- "*.py" | flake8 --diff``
 - [ ] whatsnew entry -> Should I do this for a future

### CODE REVIEW

**Comment by imih** [comment]
#16747

File: pandas/tests/api/test_api.py [hunk 1]
```diff
@@ -47,15 +47,15 @@ class TestPDApi(Base):
...
+               'SparseSeries', 'Timedelta',
                'TimedeltaIndex', 'Timestamp', 'Interval', 'IntervalIndex']
 
     # these are already deprecated; awaiting removal
     deprecated_classes = ['WidePanel', 'Panel4D',
                           'SparseList', 'Expr', 'Term']
 
     # these should be deprecated in the future
-    deprecated_classes_in_future = ['Panel']
+    deprecated_classes_in_future = ['Panel', 'TimeGrouper']
```
Comment (TomAugspurger) [inline pandas/tests/api/test_api.py]: I think `TimeGrouper` should go in the `deprecated_classes` list.

File: pandas/tests/test_resample.py [hunk 2]
```diff
@@ -3030,6 +3030,11 @@ def setup_method(self, method):
         self.ts = Series(np.random.randn(1000),
                          index=date_range('1/1/2000', periods=1000))
 
+    def test_TimeGrouper_deprecation(self):
+        # deprecation TimeGrouper, #16747
```
Comment (TomAugspurger) [inline pandas/tests/test_resample.py]: Usually we write these as # GH #16747

Comment (stevemontana1980) [inline pandas/tests/test_resample.py]: Thank, changes committed. I'm new at this, so thanks for your patience.

**Review by jreback — CHANGES_REQUESTED** [review]
- pls add a line in whatsnew (deprecations sections).
- run tests where this appears locally (e.g. the resample / groupby tests). we need to suppress this error where it is used now. so you need to change any instances of use of ``TimeGrouper`` to use ``Grouper`` (there are a few). but of course leave the tests on the deprecated one.

**Comment by jreback** [comment]
can you rebase and update according to comments.

**Comment by jreback** [comment]
can you rebase / update

**Comment by jreback** [comment]
closing in favor of #17703

**Pull request closed without merging**